crates/ere-cli/src/main.rs

```diff
@@ -0,0 +1,213 @@
+        Commands::Execute {
+            program_path,
+            input_path,
+            report_path,
+            resource,
+        } => execute(program_path, resource, input_path, report_path),
+        Commands::Verify {
+            program_path,
+            proof_path,
+        } => verify(program_path, proof_path),
+    }
+}
+
+fn compile(
+    mount_directory: PathBuf,
+    guest_relative: PathBuf,
+    program_path: PathBuf,
+) -> Result<(), Error> {
+    #[cfg(feature = "jolt")]
+    let program = ere_jolt::JOLT_TARGET::compile(&mount_directory, &guest_relative);
+
+    #[cfg(feature = "nexus")]
+    let program = ere_nexus::NEXUS_TARGET::compile(&mount_directory, &guest_relative);
+
+    #[cfg(feature = "openvm")]
+    let program = ere_openvm::OPENVM_TARGET::compile(&mount_directory, &guest_relative);
+
+    #[cfg(feature = "pico")]
+    let program = ere_pico::PICO_TARGET::compile(&mount_directory, &guest_relative);
+
+    #[cfg(feature = "risc0")]
+    let program = ere_risc0::RV32_IM_RISC0_ZKVM_ELF::compile(&mount_directory, &guest_relative);
+
+    #[cfg(feature = "sp1")]
+    let program = ere_sp1::RV32_IM_SUCCINCT_ZKVM_ELF::compile(&mount_directory, &guest_relative);
+
+    #[cfg(feature = "zisk")]
+    let program = ere_zisk::RV64_IMA_ZISK_ZKVM_ELF::compile(&mount_directory, &guest_relative);
+
+    serde::write(
+        &program_path,
+        &program.with_context(|| "Failed to compile program")?,
+        "program",
+    )
+}
+
+fn prove(
+    program_path: PathBuf,
+    resource: ProverResourceType,
+    input_path: PathBuf,
+    proof_path: PathBuf,
+    report_path: PathBuf,
+) -> Result<(), Error> {
+    let zkvm = construct_zkvm(program_path, resource)?;
+    let input = serde::read_input(&input_path)?;
+    let (proof, report) = zkvm.prove(&input).with_context(|| "Failed to prove")?;
+    fs::write(&proof_path, proof)
+        .with_context(|| format!("Failed to write proof to {}", proof_path.display()))?;
+    serde::write(&report_path, &report, "report")?;
+    Ok(())
+}
+
+fn execute(
+    program_path: PathBuf,
+    resource: ProverResourceType,
+    input_path: PathBuf,
+    report_path: PathBuf,
+) -> Result<(), Error> {
+    let zkvm = construct_zkvm(program_path, resource)?;
+    let input = serde::read_input(&input_path)?;
+    let report = zkvm.execute(&input).with_context(|| "Failed to execute")?;
+    serde::write(&report_path, &report, "report")?;
+    Ok(())
+}
+
+fn verify(program_path: PathBuf, proof_path: PathBuf) -> Result<(), Error> {
+    let zkvm = construct_zkvm(program_path, ProverResourceType::default())?;
+    let proof = fs::read(&proof_path)
+        .with_context(|| format!("Failed to read proof from {}", proof_path.display()))?;
+    zkvm.verify(&proof)
+        .with_context(|| "Failed to verify proof")?;
+    Ok(())
+}
+
+fn construct_zkvm(program_path: PathBuf, resource: ProverResourceType) -> Result<impl zkVM, Error> {
```

Comment:
Re https://github.com/eth-act/ere/pull/66#discussion_r2242919308 I think somewhere has to do this once, currently it's done in `zkevm-benchmark-workload` like https://github.com/eth-act/zkevm-benchmark-workload/blob/5e44ca4dad7be4e531730a5eab8987efbc25dc38/crates/ere-hosts/src/main.rs#L206-L258, but I think in `ere` we could do this once so the dependents don't need to do it again and again.